pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
In other words, what I was trying to say is that I think that only few people understand how our internal storage works (I mean pkg/storage + pkg/registry) and I think that's fine, since this is just an infrastructure for others.